### PR TITLE
Spells display targeted monster ids

### PR DESCRIPTION
**Pull request by John-Candlebury**
### Summary

```SUMMARY: Interface "Spells now display targeted monster ids"```

#### Purpose of change

Better display of information.

#### Describe the solution

This here is how it looks. 

image

Also take note that the repetition of names is an unfortunate result of both mon_hologram and mon_mirror_image having the same name.

### CODE REVIEW

File: src/magic.cpp [hunk 1]
```diff
@@ -1051,6 +1051,34 @@ std::string spell::enumerate_targets() const
     return ret;
 }
 
+std::string spell::list_targeted_monster_ids() const
+{
+    if( type->targeted_monster_ids.empty() ) {
+        return "";
+    }
+    std::vector<std::string> all_valid_target_ids;
+    for( auto iter : type->targeted_monster_ids ) {
```
Comment (KorGgenT) [inline src/magic.cpp]:
```suggestion
    for( const mtype_id &iter : type->targeted_monster_ids ) {
```
this isn't technically an iter, so perhaps change the variable too

File: src/magic.cpp [hunk 2]
```diff
@@ -1051,6 +1051,34 @@ std::string spell::enumerate_targets() const
...
+        all_valid_target_ids.emplace_back( iter->nname() );
+    }
+    if( all_valid_target_ids.size() == 1 ) {
+        return all_valid_target_ids[0];
+    }
+
+    std::string ret;
+    // @TODO: if only we had a function to enumerate strings and concatenate them...
```
Comment (KorGgenT) [inline src/magic.cpp]: we do have something like that. i know that for a fact. i've used it. i just can't recall what it is at this time.

File: src/magic.cpp [hunk 3]
```diff
@@ -1605,6 +1633,13 @@ void spellcasting_callback::draw_spell_info( const spell &sp, const uilist *menu
     print_colored_text( w_menu, point( h_col1, line++ ), gray, gray,
                         string_format( "%s: %s", _( "Valid Targets" ), targets ) );
 
+    std::string target_ids;
+    target_ids = sp.list_targeted_monster_ids();
+    if( !( target_ids == "" ) ) {
```
Comment (KorGgenT) [inline src/magic.cpp]:
```suggestion
    if( !target_ids.empty() ) {
```

File: src/magic.cpp (hunk 2 above)
Comment (John-Candlebury) [reply src/magic.cpp]: To be honest, I just copied that from your code above. 
But I'll search for that function just in case.

Comment (KorGgenT) [reply src/magic.cpp]: i believe that too. i'll leave you to it and look around tomorrow

Comment (Qrox) [reply src/magic.cpp]: There's `enumerate_as_string` in `output.h`.

File: src/magic.cpp [hunk 4]
```diff
@@ -1051,6 +1051,37 @@ std::string spell::enumerate_targets() const
     return ret;
 }
 
+std::string spell::list_targeted_monster_ids() const
```
Comment (Qrox) [inline src/magic.cpp]: Why not name this function `list_targeted_monster_names`? The current name gives the impression that the function returns a monster's string id.

File: src/magic.cpp [hunk 5]
```diff
@@ -1605,6 +1636,13 @@ void spellcasting_callback::draw_spell_info( const spell &sp, const uilist *menu
     print_colored_text( w_menu, point( h_col1, line++ ), gray, gray,
                         string_format( "%s: %s", _( "Valid Targets" ), targets ) );
 
+    std::string target_ids;
+    target_ids = sp.list_targeted_monster_ids();
+    if( !target_ids.empty() ) {
+        fold_and_print( w_menu, point( h_col1, line++ ), info_width, gray,
+                        string_format( "%s: %s", _( "Only affects the monsters" ), target_ids ) );
```
Comment (Qrox) [inline src/magic.cpp]:
```suggestion
                        _( "Only affects the monsters: %s" ), target_ids );
```

File: src/magic.cpp (hunk 2 above)
Comment (John-Candlebury) [reply src/magic.cpp]: Thank you Qrox!

**Pull request merged**